library/src/main/java/com/google/android/exoplayer2/metadata/id3/Id3Decoder.java

```diff
@@ -262,6 +376,37 @@ private static BinaryFrame decodeBinaryFrame(ParsableByteArray id3Data, int fram
   }
 
   /**
+   * Undo the unsynchronization applied to one or more frames.
+   * @param dataSource The original data, positioned at the beginning of a frame.
+   * @param count The number of valid bytes in the frames to be processed.
+   * @return replacement data for the frames.
+   */
+  private static ParsableByteArray removeUnsynchronization(ParsableByteArray dataSource, int count) {
+    byte[] source = dataSource.data;
+    int sourceIndex = dataSource.getPosition();
+    int limit = sourceIndex + count;
+    byte[] dest = new byte[count];
+    int destIndex = 0;
+
+    while (sourceIndex < limit) {
+      byte b = source[sourceIndex++];
+      if ((b & 0xFF) == 0xFF) {
+        int nextIndex = sourceIndex+1;
```

Comment:
This looks wrong to me. sourceIndex is incremented on L392 already, so nextIndex ends up being two greater than the index at which 0xFF is found. It should be one greater, I think.